proposals/model-management.md

```diff
@@ -0,0 +1,149 @@
+
+While working within the following constraints:
+
+* KubeAI should be able to be installed with a single command
+* KubeAI should not rely on external dependencies to provide basic functionality
+* KubeAI should be able to run on a local machine
+* KubeAI should be able to be run on clouds with heterogenous disk features (limited support for ReadOnlyMany)
+
+## Solution
+
+KubeAI should support configurable storage for both models and adapters.
+
+These settings should be configurable on a per Model basis because the optimum caching technique is relative to the size and frequency-of-use of the model. For example: GCP Hyperdisk ML is relatively expensive compared to other options, however it can be highly performant especially when it comes to loading large models. They should also be configurable on a per-finetune-job basis so that users can use KubeAI to publish adapters to their huggingface repos.
+
+### Example Models
+
+```yaml
+# Model configured with adapters from huggingface
+kind: Model
+spec:
+  url: "hf://meta-llama/Meta-Llama-3.1-8B-Instruct"
+  adapters:
+  # Adapter can be requested by name at inference time
+  - name: sarcasm
+    url: "hf://myrepo/sarcastic-lora-adapter-for-llama-3-8b-Instruct"
+
+# Model that explicitly asks for a cache profile
+kind: Model
+spec:
+  url: "hf://meta-llama/Meta-Llama-3.1-405B-Instruct"
+  cacheProfile: HyperDisk
+
+# A Model without any cache specified will use the default set in KubeAI's config.yaml
+kind: Model
+spec:
+  url: "hf://meta-llama/Meta-Llama-3.1-8B-Instruct"
+
+# Model that explicitly wont use a cache
+kind: Model
+spec:
+  url: "hf://meta-llama/Meta-Llama-3.1-8B-Instruct"
+  cacheProfile: None
+```
+
+### Finetuning Example
+
+If a user submits a finetuning request (`POST /v1/fine_tuning/jobs`), they have the option to specify a location to store the adapters:
+
+```json
+# OpenAI-compatible finetune job request
+{
+  # Model.metadata.name
+  "model": "llama-3.1-8b",
+  # KubeAI-Only: Training file could be specified as URL: "s3://..."
+  "training_file": "<id-of-file-uploaded-via-files-api>",
+  # KubeAI-Only: Location to store parameters
+  # Alternative: "hf://some-user/some-repo"
+  # If not specified, default location determined from KubeAI config.yaml.
+  "storage_url": "s3://some-bucket/some-dir/"
+}
+```
+
+After this job completes the base Model would have an adapter appended:
+
+```yaml
+kind: Model
+spec:
+  url: "hf://meta-llama/Meta-Llama-3.1-8B-Instruct"
+  adapters:
+  # +Adapter:
+  - name: "name-of-finetune-job"
+    url: "s3://some-bucket/some-dir/name-of-finetune-job"
+```
+
+Users can specify an adapter by name at inference time.
+
+### KubeAI Config
+
+Model cache profiles and finetune default storage locations should be defined at the KubeAI `config.yaml` level:
+
+```yaml
+finetuning:
+  storage:
+    defaultURLTemplate: "s3://my-bucket/finetuning/{model}/{job}"
+  
+models:
+  cache:
+    defaultProfile: Bucket # Could also be "None"
```

Comment:
if not specified I'm assuming the helm default values would look like this:
```
models:
  cache:
    defaultProfile: ""
    profiles: {}
```

Reply:
Maybe more explicit with None:

```yaml
models:
  cache:
    defaultProfile: None
    profiles: {}
```